subprojects/docs/src/snippets/buildCache/integration-tests/groovy/build.gradle

```diff
@@ -27,38 +27,39 @@ tasks.named('integTest') {
 // end::distributionPathInput[]
 
 // tag::distributionDirInput[]
-class DistributionLocationProvider implements CommandLineArgumentProvider { // <1>
+class DistributionLocationProvider implements CommandLineArgumentProvider {  // <1>
     @InputDirectory
-    @PathSensitive(PathSensitivity.RELATIVE)                                // <2>
-    File distribution
+    @PathSensitive(PathSensitivity.RELATIVE)  // <2>
+    Provider<Directory> distribution
```

Comment:
🤔 Shouldn't this be `final`?

Reply:
I guess another, more idiomatic way would be to have this as a `Property` and then set its value (from the provider that is currently passed in the constructor).

Reply:
> Shouldn't this be final?

It cannot be, unless I also write a proper constructor or use explicit `MapConstructor` annotation.

>I guess another, more idiomatic way would be to have this as a Property and then set its value (from the provider that is currently passed in the constructor).

This involves using `ObjectFactory`, though it is probably fine - no need to actually instantiate the `CommandLineArgumentProvider` with it, just use `ObjectFactory.property`.

Reply:
> You can also instantiate `DistributionLocationProvider` itself via `ObjectFactory`, in which case it can have an abstract getter for the property.

It isn't as verbose as I initially estimated, so let's go with it. I've also added a link to the "Custom Gradle types" chapter in a footnote, so users can learn more about this magic.